### ISSUE TITLE
Developer reply emails for content-reviewers link to technical-review page

### ISSUE THREAD

**Issue by wagnerand** [mozilla/addons-server#6936]
This is a bit of a hard bug but maybe someone has an idea for an elegant solution.

STR:
1. Do a content-review. Either content-reject or send an information request.
2. The developer replies.
3. The email that is sent to the content reviewer contains the link to the technical review page, to which they usually don't have access.

I think what needs to happen is when receiving the email from the developer (after step 2) we need to determine whether it was a content-review (rejection or information request) and adapt the link in the email we send out to reviewers (in step 3).

**Comment by diox** [comment]
Dumb solution: include both links?

**Comment by wagnerand** [comment]
Ok, I'll take 'dumb' solutions as well! 😸 

On a more serious note, I think that should work for the time being.

I heard there is a rewrite coming up, must be any ~~day~~ quarter now.

**Comment by kewisch** [comment]
Another thing we could do that might be low effort is to redirect review-listed to review-content if the only permissions the reviewer has are content review.

**Comment by wagnerand** [comment]
As mentioned on IRC, this only works as long as technical reviewers and content reviewers are entirely distinct groups, which might not be the case in the near future.

**Comment by bjherbison** [comment]
Any possibility of the dumb solution being implemented? There has been no activity in over a year and this issue causes me extra work multiple times a day.

The change should go somewhere around line 390 in src/olympia/activity/utils.py , although it might be best to use a new to_reviewer.txt split from src/olympia/activity/templates/activity/emails/developer.txt since there will be an additional field.

**Comment by bjherbison** [comment]
The situation has changed. The link in the message is now readable by content reviewer but we still need to change the URL to create a response or take an action.

**Comment by bjherbison** [comment]
The change noted in September has been reverted. The link in the e-mail now takes me to a page that I can't read so I need to change the URL before I can see any useful information.

As I noted last year, this issue causes me extra work multiple times a day.

**Comment by diox** [comment]
@bjherbison you should now be able to view the technical review page and reply there. Can you confirm ?

**Comment by bjherbison** [comment]
I can now see the /review-listed/ version of pages, but replying gives me "Oops! Not allowed.".

And replying would only be a partial solution as I would still need to edit the URL to content-approve or reject as those two options don't show up for me on the /review-listed/ version of pages.

**Comment by bjherbison** [comment]
Actually, I can reply. It now gives me the error after posting the reply.

But again, this is still no a full solution to the original problem.

I thought Caitlin proposed that the redirection solution above be implemented.
https://github.com/mozilla/addons-server/issues/6936#issuecomment-346573683

**Comment by diox** [comment]
Yeah, the error is just on the queue page you're redirected to. I know it's not a full solution, I'm working on a full fix.

**Comment by diox** [comment]
QA: the link is staying the same, but now content reviewers that don't have other permissions allowing them to review add-ons will get redirected to the content review page when they access the regular review page for an add-on.

**Comment by ioanarusiczki** [comment]
@diox The e-mails send the link of this format:
https://addons-dev.allizom.org/en-US/firefox/addon/facebook-containertest/ 
With a content reviewer account the URL redirects to:
https://reviewers.addons-dev.allizom.org/en-US/reviewers/review-content/facebook-containertest

![redirect](https://user-images.githubusercontent.com/33448286/78124807-e21fe680-7418-11ea-980a-9b53acb2a58c.gif)


as a note:
These e-mails from developers with their replies are sent to the reviewer only when the content review of the last version is rejected (or all versions). E-mails are not sent if the reviewer rejects the content of a previous approved version and the developer decides to write something from the versions page https://addons-dev.allizom.org/en-US/developers/addon/facebook-containertest/versions
Let me know if that's alright so I can mark this issue verified.

**Comment by bjherbison** [comment]
The last comment isn't correct. The e-mail messages I receive have links with this format:

https://addons.mozilla.org/reviewers/review-listed/2632072

And the messages are sent whenever a developer replies, independent of whether the add-on has been content rejected, or rejected, or not.

**Comment by ioanarusiczki** [comment]
@bjherbison Yes, was my mistake , the URLs contain the add-on id (and not the slug as I mentioned in my comment). 
In this case, I can mark this verified on FF74(Win10). Probably what I've described above is not a real case scenario, if it was it would have been filed a long time ago.

**Comment by bjherbison** [comment]
When is this change supposed to go to production?

**Comment by wagnerand** [comment]
> When is this change supposed to go to production?

This is going to be on production later today, see the _Milestone_ info on the right side on this issue page.

**Comment by bjherbison** [comment]
Thanks, when I saw "verified" I expected it to be live. (I'm used to the process flow for other tracking systems.)

I just checked and the fix is live. Links that were an annoyance to me yesterday are convenient now. Thank you!

**Comment by bjherbison** [comment]
This fix has regressed.

Today I received an e-mail with the standard link: 
https://addons.mozilla.org/reviewers/review-listed/#######
Clicking the link I was taken to:
https://reviewers.addons.mozilla.org/en-US/reviewers/review-content/#######
but not to:
https://reviewers.addons.mozilla.org/en-US/reviewers/review-content/name_of_add-on

Previously the e-mail links would redirect to the "name_of_add-on". From the "#######" URL I comment but I can't approve, redirect, "Reviewer reply", or "Request super-review".

Should this issue be reopened or should a new issue be created?

**Comment by diox** [comment]
Please open a new issue and give the exact URLs present in the mail and that you were redirected to.

**Comment by bjherbison** [comment]
I'm sorry, it was a false alert. The author has been creating and deleting multiple add-ons with the same name (at least three deleted).

I checked other recent e-mail messages and they all worked.
